postprocessor/src/main/java/com/azure/autorest/postprocessor/Postprocessor.java

```diff
@@ -26,16 +40,25 @@ public Postprocessor(Connection connection, String plugin, String sessionId) {
     super(connection, plugin, sessionId);
   }
 
+  @SuppressWarnings("unchecked")
   @Override
   public boolean processInternal() {
     List<String> files = listInputs();
     Map<String, String> fileContents = files.stream().collect(Collectors.toMap(f -> f, this::readFile));
 
+    String jarPath = JavaSettings.getInstance().getCustomizationJarPath();
+    String className = JavaSettings.getInstance().getCustomizationClass();
+    Map.Entry<String, String> readme = getReadme();
+
+    if (className == null || (jarPath == null && readme == null)) {
+      logger.warn("Must provide a JAR path or a README.md config containing the customization class {}", className);
+      return false;
+    }
+
     try {
       //Step 1: post process
-      String jarPath = JavaSettings.getInstance().getCustomizationJarPath();
-      String className = JavaSettings.getInstance().getCustomizationClass();
-      if (jarPath != null && className != null) {
+      Class<? extends Customization> customizationClass;
+      if (jarPath != null) {
```

Comment:
Could there be a security risk if jarUrl can be loaded from remote?

Reply:
Can you elaborate on the risk here? It can be loaded from anywhere over the Internet, but it's passed in by the user and it's the user's responsibility to make sure they don't load random jars from untrusted sources.

Reply:
Actually, I do not exactly know. Theoretically someone can modify the url hiding it among other changes in a submitted PR, trick dev to test the PR and hence run the jar, and the jar would grab env and pass file from local disk, upload them to remote site. Will it happen? No idea. The risk probably is not significant at this stage. Does it worth some mitigation e.g. SecurityManager, or just disable remote jar? Not sure.

Reply:
I think we should phase out passing in a separate jar - and use either the readme or a central location in the repo. So I'll not do too much here